source/basic.tex

```diff
@@ -1777,13 +1777,15 @@
 such a name undergoes unqualified name lookup from the point where it appears.
 
 \pnum
-An unqualified name that is a component name\iref{expr.prim.id.unqual} of
+Lookup for an unqualified name $U$
+that is a component name\iref{expr.prim.id.unqual} of
 a \grammarterm{type-specifier} or \grammarterm{ptr-operator} of
-a \grammarterm{conversion-type-id} is looked up in the same fashion
-as the \grammarterm{conversion-function-id} in which it appears.
-If that lookup finds nothing, it undergoes unqualified name lookup;
-in each case, only names
-that denote types or templates whose specializations are types are considered.
+a \grammarterm{conversion-type-id} $T$ considers only names
+that denote types or templates whose specializations are types.
+If $T$ appears in a \grammarterm{conversion-function-id} that is a qualified name
+whose lookup context\iref{basic.lookup.qual} is a class $S$,
+lookup for $U$ performs a search in the scope associated with $S$;
+if that lookup finds nothing, $U$ undergoes unqualified name lookup.
```

Comment:
I think it would be clearer to say "[&hellip;]; **otherwise, or** if that lookup finds nothing, [&hellip;]".  It's true that the general "Unless otherwise specified" from the previous paragraph ought to apply, but it tends to sound like the alternative is that the name is not looked up at all.

Reply:
> It's true that the general "Unless otherwise specified" from the previous paragraph ought to apply, but it tends to sound like the alternative is that the name is not looked up at all.

To me, the previous paragraph clearly applies to unqualified `\grammarterm{conversion-function-id}` :/
May I not apply the proposed change, please?

Reply:
This paragraph is about an unqualified _U_, looked up separately from the *conversion-function-id* that contains it.  I want to avoid the (nonsensical) suggestion that, if _T_ does not appear in the given context, _U_ is not looked up at all.

Reply:
> This paragraph is about an unqualified _U_, looked up separately from the _conversion-function-id_ that contains it.

Yeah, right, but whatever, the previous paragraph applies to other unqualified names (which are not mentioned in this paragraphs) (and in any other paragraphs-exceptions).

> I want to avoid the (nonsensical) suggestion that, if _T_ does not appear in the given context, _U_ is not looked up at all.

I don't see how is it possible. I mean, not only the previous paragraph, but also the previous sentence suggest that such _U_ are looked up:
> Lookup for an unqualified name _U_ that is ... of a _conversion-type-id_ _T_ considers only names that ... If _T_ appears in a _conversion-function-id_ ...

If _U_ is not looked up when _T_ doesn't appear in _conversion-function-id_, there is no reason these sentences are in this and not the opposite order

Reply:
I don't insist on a change; it just strikes me as a trip hazard for the reader to say "If X, then Y; if Y does nothing, Z." and have Z also happen when X doesn't pertain.